app/Http/Controllers/SportsController.php

```diff
@@ -2,31 +2,76 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\SportsStoreRequest;
+use App\Models\Medal;
 use App\Models\Sport;
 use App\Models\Country;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Str;
 
 class SportsController extends Controller
 {
     public function create()
     {
         $sports = Sport::all();
         $countries = Country::all();
+        $medal_names = Medal::NAMES;
 
-        return view('sports.create', compact('sports', 'countries'));
+        return view('sports.create', compact('sports', 'countries', 'medal_names'));
     }
 
-    public function store(Request $request)
+    public function store(SportsStoreRequest $request)
     {
-        // Add your code here
+        // Added my own code here
+        // return $request->all();
+
+        DB::transaction(function () use ($request) {
```

Comment:
Transactions for such simple task I think is overkill

Reply:
I inserted value using loop. In a loop, there could be an occur in any iteration. For safety, I used db transaction.

Reply:
@krekas in this case, I disagree. Inside of that foreach loop, a lot of things may go wrong, so transaction would be rolled back.

However, @shibbirweb your loop is way too overcomplicated, so it increases the probability of bugs. I will shoot a video with my version of how to implement everything, and you would compare with your solution.

Reply:
@PovilasKorop Thanks for reviewing my code.
First time implemented manual process. There was no for each loop instead used hardcoded value. Then I thought to make it dynamic however It was not necessary. Then I use constant of three positions (gold, silver, bronze) and implement in this way.

Reply:
@PovilasKorop This was my first approach. There was a similar code in each position. Then I refactored into that way

```
// for 1st position
foreach ($request->gold as $sport_id => $country_id) {
    $country_query = Sport::find($sport_id)
        ->ownMedalsCountries();

    $existed_country = $country_query
        ->find($country_id);

    // check country already own any medals
    if ($existed_country) {
        // increment medal value
        $existed_country->medals->increment('gold');
    } else {
        // not owned any medals
        // attach new medal value
        $country_query->attach($country_id, ['gold' => 1]);
    }
}


// for 2nd position
foreach ($request->silvers as $sport_id => $country_id) {
    $country_query = Sport::find($sport_id)
        ->ownMedalsCountries();

    $existed_country = $country_query
        ->find($country_id);

    // check country already own any medals
    if ($existed_country) {
        // increment medal value
        $existed_country->medals->increment('silvers');
    } else {
        // not owned any medals
        // attach new medal value
        $country_query->attach($country_id, ['silver' => 1]);
    }
}


// for 3rd position
foreach ($request->bronzes as $sport_id => $country_id) {
    $country_query = Sport::find($sport_id)
        ->ownMedalsCountries();

    $existed_country = $country_query
        ->find($country_id);

    // check country already own any medals
    if ($existed_country) {
        // increment medal value
        $existed_country->medals->increment('bronze');
    } else {
        // not owned any medals
        // attach new medal value
        $country_query->attach($country_id, ['bronze' => 1]);
    }
}
```